edx_oauth_client/pipeline.py

```diff
@@ -1,139 +1,72 @@
 import logging
-from md5 import md5
 
 from django.contrib.auth.models import User
-from django.http import HttpResponseBadRequest
-from django.template.defaultfilters import slugify
+from social_core.pipeline import partial
+from third_party_auth.pipeline import AuthEntryError, is_api, get_complete_url
 
-from social.pipeline import partial
+from openedx.core.djangoapps.user_authn.views.registration_form import AccountCreationForm
+from openedx.core.djangoapps.user_authn.utils import generate_password
+from student.helpers import do_create_account
 
-from student.views import create_account_with_params, reactivation_email_for_user
-from student.models import UserProfile
-from third_party_auth.pipeline import AuthEntryError, make_random_password
 
 log = logging.getLogger(__name__)
 
-# The following are various possible values for the AUTH_ENTRY_KEY.
-AUTH_ENTRY_LOGIN = 'login'
-AUTH_ENTRY_REGISTER = 'register'
-AUTH_ENTRY_ACCOUNT_SETTINGS = 'account_settings'
-
-AUTH_ENTRY_LOGIN_2 = 'account_login'
-AUTH_ENTRY_REGISTER_2 = 'account_register'
-
-# Entry modes into the authentication process by a remote API call (as opposed to a browser session).
-AUTH_ENTRY_LOGIN_API = 'login_api'
-AUTH_ENTRY_REGISTER_API = 'register_api'
-
 
 @partial.partial
 def ensure_user_information(
-        strategy, auth_entry, backend=None, user=None, social=None,
-        allow_inactive_user=False, *args, **kwargs):
+        strategy, auth_entry, backend=None, user=None, social=None, allow_inactive_user=False, *args, **kwargs
+):
     """
-    Ensure that we have the necessary information about a user (either an
-    existing account or registration data) to proceed with the pipeline.
+    Ensure that we have the necessary information about a user to proceed with the pipeline.
+
+    Either an existing account or registration data.
     """
 
-    response = {}
     data = {}
-
     try:
-        if 'data' in kwargs['response']:
-            user_data = kwargs['response']['data'][0]
+        if "data" in kwargs["response"]:
+            user_data = kwargs["response"]["data"][0]
```

Comment:
I'm curious are we sure about kwargs['response'] data type? and expected key availability as well about kwargs has the `'response'` key? Any uncertainty will bring an error.